x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java

```diff
@@ -403,9 +403,9 @@ public static Path resolveConfigFile(Environment env, String name) {
      * of SSLContexts when configuration files change on disk.
      */
     private SSLService createSSLService(Environment environment, ResourceWatcherService resourceWatcherService) {
-        final Map<String, SSLConfiguration> sslConfigurations = SSLService.getSSLConfigurations(environment.settings());
+        final Map<String, SslConfiguration> sslConfigurations = SSLService.getSSLConfigurations(environment);
         final SSLConfigurationReloader reloader =
-            new SSLConfigurationReloader(environment, resourceWatcherService, sslConfigurations.values());
+            new SSLConfigurationReloader(resourceWatcherService, sslConfigurations.values());
```

Comment:
The ssl-config code takes an environment (actually `Path basePath`) in the constructor, rather than having it passed to each method.
A lot of the changes in this PR are about removing `environment` where it isn't needed anymore.